openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TestCaseRepository.java

```diff
@@ -790,6 +790,7 @@ public TestCase addFailedRowsSample(
             "failedRowsSample",
             JsonUtils.pojoToJson(tableData));
     setFieldsInternal(testCase, Fields.EMPTY_FIELDS);
+    testCase = JsonUtils.readValue(JsonUtils.pojoToJson(testCase), TestCase.class);
```

Comment:
isn't `testCase` already of type `TestCase`?

Reply:
@sushi30  you can directly do JsonUtils.deepCopy

Reply:
it is. the object needs to be deep copied to avoid modifying the object in the cache

Reply:
changed to use JsonUtils.deepCopy